### PR TITLE
Implement anyhow context for artifact builder

### PR DESCRIPTION
**Pull request by dlachaumepalo**
## Content
This PR includes an update of mithril-aggregator `artifact_builder` with `anyhow` context.

## Pre-submit checklist

- Branch
  - [ ] Tests are provided (if possible)
  - [x] Crates versions are updated (if relevant)
  - [X] Commit sequence broadly makes sense
  - [X] Key commits have useful messages
- PR
  - [x] No clippy warnings in the CI
  - [X] Self-reviewed the diff
  - [X] Useful pull request description
  - [x] Reviewer requested
- Documentation
  - [ ] Update README file (if relevant)
  - [ ] Update documentation website (if relevant)
  - [ ] Add dev blog post (if relevant)

## Issue(s)

Relates to #798

### CODE REVIEW

File: mithril-aggregator/src/artifact_builder/cardano_immutable_files_full.rs [hunk 1]
```diff
@@ -24,7 +25,7 @@ use mithril_common::{
 pub enum CardanoImmutableFilesFullArtifactError {
     /// Protocol message part is missing
     #[error("Missing protocol message: '{0}'.")]
-    MissingProtocolMessage(String),
+    MissingProtocolMessage(StdError),
```
Comment (jpraynaud) [inline mithril-aggregator/src/artifact_builder/cardano_immutable_files_full.rs:28]: This is maybe more straightforward:
```suggestion
    MissingProtocolMessage(Beacon),
```

File: mithril-aggregator/src/artifact_builder/mithril_stake_distribution.rs [hunk 2]
```diff
@@ -7,18 +8,17 @@ use super::ArtifactBuilder;
...
 #[derive(Debug, Error)]
 pub enum MithrilStakeDistributionArtifactBuilderError {
-    /// Could not get protocol parameters
-    #[error(
-        "Could not build Mithril Stake Distribution artifact, no protocol parameters available."
-    )]
-    NoProtocolParameters(),
+    /// Protocol parameters are missing
+    #[error("Missing protocol parameter: '{0}'.")]
+    MissingProtocolParameters(StdError),
```
Comment (jpraynaud) [inline mithril-aggregator/src/artifact_builder/mithril_stake_distribution.rs:19]: Same suggestion as above:
```suggestion
    MissingProtocolParameters(Epoch),
```

**Review by ghubertpalo — APPROVED** [review]

File: mithril-aggregator/src/artifact_builder/cardano_immutable_files_full.rs [hunk 3]
```diff
@@ -24,7 +25,7 @@ use mithril_common::{
 pub enum CardanoImmutableFilesFullArtifactError {
     /// Protocol message part is missing
     #[error("Missing protocol message: '{0}'.")]
```
Comment (jpraynaud) [inline mithril-aggregator/src/artifact_builder/cardano_immutable_files_full.rs:27]:
```suggestion
    #[error("Missing protocol message for beacon: '{0}'.")]
```

File: mithril-aggregator/src/artifact_builder/mithril_stake_distribution.rs [hunk 4]
```diff
@@ -13,12 +13,11 @@ use mithril_common::{
 /// Error linked to [MithrilStakeDistributionArtifactBuilder].
 #[derive(Debug, Error)]
 pub enum MithrilStakeDistributionArtifactBuilderError {
-    /// Could not get protocol parameters
-    #[error(
-        "Could not build Mithril Stake Distribution artifact, no protocol parameters available."
-    )]
-    NoProtocolParameters(),
+    /// Protocol parameters are missing
+    #[error("Missing protocol parameter: '{0}'.")]
```
Comment (jpraynaud) [inline mithril-aggregator/src/artifact_builder/mithril_stake_distribution.rs:17]:
```suggestion
    #[error("Missing protocol parameters for epoch: '{0}'.")]
```

**Review by jpraynaud — APPROVED** [review]
LGTM :+1: 

Just 2 typos to fix before merging

**Pull request merged**